src/Swashbuckle.AspNetCore.SwaggerGen/SwaggerGenerator/SwaggerGeneratorOptions.cs

```diff
@@ -61,12 +65,22 @@ public SwaggerGeneratorOptions()
 
         public IList<IParameterFilter> ParameterFilters { get; set; }
 
+        public IList<IParameterAsyncFilter> ParameterAsyncFilters { get; set; }
+
         public List<IRequestBodyFilter> RequestBodyFilters { get; set; }
 
+        public List<IRequestBodyAsyncFilter> RequestBodyAsyncFilters { get; set; }
+
         public List<IOperationFilter> OperationFilters { get; set; }
 
+        public List<IOperationAsyncFilter> OperationAsyncFilters { get; set; }
+
         public IList<IDocumentFilter> DocumentFilters { get; set; }
 
+        public IList<IDocumentAsyncFilter> DocumentAsyncFilters { get; set; }
+
+        public bool HasAsyncFilters => (ParameterAsyncFilters?.Any() ?? false) || (RequestBodyAsyncFilters?.Any() ?? false) || (OperationAsyncFilters?.Any() ?? false) || (DocumentAsyncFilters?.Any() ?? false);
```

Comment:
I don't we need this - they shouldn't be mutually exclusive.

Reply:
Ideally people should be able to use both if we can design it that way.

Reply:
I don't feel super-strongly about this, just let me know which option you prefer:

1. GetSwagger() silently ignores async filters
2. GetSwagger() throws error if async filters are used

Reply:
> Ideally people should be able to use both if we can design it that way.

Sorry, I missed this comment before my last answer. What are you referring to with `both`? `GetSwagger()` and `GetSwaggerAsync()` or async and non-async filters?

Reply:
Just the filters - if there's an async middleware, then it could use both implementations, where as a sync middleware/interface/whatever would only be able to use the sync ones - in that case it should probably just ignore the async ones entirely.

Reply:
Ok, I will implement that.